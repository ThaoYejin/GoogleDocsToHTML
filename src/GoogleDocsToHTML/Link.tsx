import { docs_v1 } from "googleapis";
import React from "react";
import { TextRun } from "./TextRun";


export interface LinkProps {
    paragraphElement: docs_v1.Schema$ParagraphElement | undefined;
}

export const Link: React.FC<LinkProps> = ({
    paragraphElement
}) => {
    var url: string | undefined;
    if (paragraphElement?.textRun?.textStyle?.link != undefined) {
        url = paragraphElement?.textRun?.textStyle?.link.url;
    }

    return (
        <a href={url}>
            <TextRun element={paragraphElement}></TextRun>
        </a>
    )
}