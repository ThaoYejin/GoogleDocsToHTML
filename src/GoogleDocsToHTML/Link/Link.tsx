import { docs_v1 } from "googleapis";
import React from "react";
import { TextRun } from "../TextRun/TextRun";


export interface LinkProps {
    paragraphElement: docs_v1.Schema$ParagraphElement | undefined;
}

export const Link: React.FC<LinkProps> = ({
    paragraphElement
}) => {
    var url: string | undefined;

    //Get the URL of the link
    if (paragraphElement?.textRun?.textStyle?.link != undefined) {
        url = paragraphElement?.textRun?.textStyle?.link.url!;
    }

    return (
        <a href={url}>
            <TextRun element={paragraphElement}></TextRun>
        </a>
    )
}