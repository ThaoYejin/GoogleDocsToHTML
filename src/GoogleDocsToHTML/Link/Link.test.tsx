import React from 'react';
import renderer from 'react-test-renderer';
import { Link } from './Link';

it('renders correctly', () => {
    var paragraphElement = {
        "startIndex": 1,
        "endIndex": 7,
        "textRun": {
            "content": "Google",
            "textStyle": {
                "underline": true,
                "foregroundColor": {
                    "color": {
                        "rgbColor": {
                            "red": 0.06666667,
                            "green": 0.33333334,
                            "blue": 0.8
                        }
                    }
                },
                "link": {
                    "url": "https://www.google.com/"
                }
            }
        }
    };

    const tree = renderer
        .create(<Link paragraphElement={paragraphElement} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
})