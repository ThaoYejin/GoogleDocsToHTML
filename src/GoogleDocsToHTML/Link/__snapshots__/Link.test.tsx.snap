// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<a
  href="https://www.google.com/"
>
  <span
    style={
      Object {
        "color": "rgb(17.00000085,85.0000017,204)",
        "textDecoration": "underline",
      }
    }
  >
    Google
  </span>
</a>
`;
